workshop/exercise-1.md

```diff
@@ -0,0 +1,228 @@
+Las técnicas de prompting (o ingeniería de prompts) son el arte y la ciencia de crear entradas (prompts) efectivas para guiar a los Modelos de Lenguaje Grandes (LLMs) a generar las respuestas deseadas. La forma en que se formula una pregunta o una instrucción puede cambiar drásticamente la calidad, relevancia y formato de la salida del modelo.
+
+Existen varias técnicas, como:
+- __Zero-shot prompting (de cero disparos)__: Pedir al modelo que realice una tarea sin darle ningún ejemplo.
+- __Few-shot prompting (de pocos disparos)__: Proporcionar al modelo algunos ejemplos de la tarea para guiar su respuesta.
+- __Chain-of-Thought (Cadena de Pensamiento)__: Animar al modelo a que "piense en voz alta" y desglose su razonamiento paso a paso.
+
+En este ejercicio, nos centraremos en el Zero-shot prompting, la técnica más fundamental. Consiste en formular una solicitud directa al LLM, confiando en su conocimiento preexistente y su capacidad de generalización para entender y ejecutar la tarea sin ejemplos previos. Por ejemplo, pedirle que traduzca un texto o resuma un artículo sin mostrarle cómo hacerlo.
+
+Este primer ejercicio tiene como objetivo introducir la interacción con un LLM usando Spring AI, implementando un caso de uso de prompt simple.
+
+## Manos a la obra
+
+Crea una clase que implemente la interfaz `ChatAssistant`, llamada `ChatAssistantService`. 
+
+### Parte 1 - Añadir el objeto ChatClient
+
+Usaremos un objeto `ChatClient` para interactuar con el LLM. Este objeto se puede construir con `ChatClient.Builder`, que ya está instanciado gracias a la autoconfiguración de Spring Boot.
+
+Crea un atributo privado y final de tipo `ChatClient` llamado `chatClient`.
+Escribe el constructor de la clase que creaste previamente `ChatAssistantService` con los parámetros `ChatClient.Builder builder` y `@Value("classpath:/system-prompt.md") Resource systemPrompt`.
+
+Asigna `this.chatClient` con el resultado de llamar `builder.defaultSystem(systemPrompt).build()`.
+
+```java
+import org.springframework.stereotype.Service;
+
+private final ChatClient chatClient;
+
+@Service
+public ChatAssistantService(ChatClient.Builder builder, @Value("classpath:/system-prompt.md") Resource systemPrompt) {
+   this.chatClient = builder
+           .defaultSystem(systemPrompt)
+           .build();
+}
+```
+
+### Parte 2 - Completar el System Prompt
+
+Actualiza el archivo `system-prompt.md` en la carpeta `src/main/resources` con el siguiente contenido:
+
+```markdown
+Responde la pregunta lo más breve posible, en español.
+```
+
+### Parte 3 - Configurar el model desde `application.properties`
+
+En lugar de configurar las opciones del modelo directamente en el código, usaremos el archivo `application.properties`. Esto nos brinda una configuración más flexible y centralizada, y nos permite cambiar el comportamiento del modelo sin tener que recompilar la aplicación.
+
+Añade las siguientes propiedades al final de tu archivo `src/main/resources/application.properties`:
```

Comment:
If I remember correctly, the file does not yet exist in this branch. Should we add it and keep it empty or prompt them to create it?